bio_hansel/kmer_val_calc/__init__.py

```diff
@@ -0,0 +1,33 @@
+from bio_hansel.kmer_val_calc.kmer_utils import calc_error_rate, calc_avg_kmer_depth
+from scipy.stats import norm as stats
+
+from bio_hansel.subtyping_params import SubtypingParams
+
+
+def find_min_kmer_val(hist: dict, subtyping_params: SubtypingParams) -> float:
+    """Find the minimum kmer frequency coverage value from observation
+    Note:
+            This method will find three things:
+            1) Error Rate
+            2) The average k-mer coverage depth
+            3) The minimum kmer cut off value
+
+            How does this work?
+            1) Estimate the expected k-mer coverage depth from observation.
+            2) (Slightly under-) estimate the k-mer error rate by assuming k-mers that appear once are errors.
+            3) Multiply those two things together to get the (slightly underestimated)
+            expected k-mer coverage depth of errors.
+            4) Plug the expected k-mer coverage depth of errors into Poisson to pull out the minimum depth
+            to be confident "the observation is not entirely cause by errors".
+
+    :arg:
+            :param hist: Dictionary containing the histogram of the observed kmer values.
+            :param subtyping_params: Parameters containing savgol and percent confidence values.
+
+    :returns:
+            The minimum kmer cut off value.
+    """
+    error_rate = calc_error_rate(hist)
+    kmer_cov_freq = calc_avg_kmer_depth(hist, subtyping_params)
+    min_kmer_val = stats.ppf(subtyping_params.kmer_cov_perc_confidence, kmer_cov_freq * error_rate)
```

Comment:
This function seems to return a very high k-mer value if you calculate the error rate with the proportion of kmers that are observed once (min kmer threshold = 51 vs 78 mean kmer coverage) vs twice (min kmer threshold = 3). Would that be why you went with the kmers that were observed twice?

Reply:
I used k-mers that are observed twice as when running `jellyfish histo` the first key while processing the file is of value `2`.  There is no `1` that results from running jellyfish histo.
ex:
me@me:~/workspace/bio_hansel$ jellyfish histo '/tmp/20171205143218-bio_hansel-SRR5646583_SMALL_fastq/2017Dec05_14_32_18-SRR5646583_SMALL.fastq.jf'
2 28149
4 2811
6 2452
8 2292
10 2319
12 2141
14 2268
16 2404
18 2190
20 2389
22 2133
24 1956
26 2034
28 1917
30 1956
32 1880
34 1532
36 1327
38 1268
40 1191
42 1122
44 768
46 700
48 724
50 597
52 591
54 494
56 493
58 378
60 299
62 218
64 144
66 152
68 90
70 60
72 39
74 37
76 48
78 8
80 19
82 16
84 8
86 13
88 6
90 10
92 1

Reply:
Did you try with a real read set? Because I get a ton of kmers only observed once.

e.g. with SRR5646583
```
⋊> jellyfish histo mer_counts.jf
1 21168363                                                                                                                                                                      
2 636551                                                                                                                                                                      
3 126247                                                                                                                                                                                                                                      
4 49265
5 24331
6 14650
7 9604
8 6698
9 5150
10 4537
11 3974
12 3898
13 3859
14 4090
15 3601
...
```

Reply:
Perhaps it was my filtered dataset that was causing issues?
Running with `-f` shows:
```
me@me:~/workspace/bio_hansel$ jellyfish histo -f '/tmp/20171205143218-bio_hansel-SRR5646583_SMALL_fastq/2017Dec05_14_32_18-SRR5646583_SMALL.fastq.jf'
0 0
1 0
2 28149
3 0
4 2811
5 0
6 2452
7 0
8 2292
9 0
```